src/main/scala/it/pagopa/interop/apigateway/service/AttributeRegistryProcessService.scala

```diff
@@ -1,14 +1,31 @@
 package it.pagopa.interop.apigateway.service
 
-import it.pagopa.interop.attributeregistryprocess.client.model.{Attribute, AttributeSeed}
+import it.pagopa.interop.attributeregistryprocess.client.model.{Attribute, Attributes, AttributeSeed}
 
 import java.util.UUID
-import scala.concurrent.Future
+import scala.concurrent.{Future, ExecutionContext}
 
 trait AttributeRegistryProcessService {
 
   def getAttributeById(attributeId: UUID)(implicit contexts: Seq[(String, String)]): Future[Attribute]
-  def getBulkAttributes(attributeIds: Set[UUID])(implicit contexts: Seq[(String, String)]): Future[Seq[Attribute]]
+  def loadBulkAttributes(
+    attributeIds: Set[UUID]
+  )(implicit contexts: Seq[(String, String)], ec: ExecutionContext): Future[Seq[Attribute]] = {
+
+    def getAttributesFrom(offset: Int): Future[Seq[Attribute]] =
+      getBulkAttributes(attributeIds = attributeIds, offset = offset, limit = 50).map(_.results)
+
+    def go(start: Int)(as: Seq[Attribute]): Future[Seq[Attribute]] =
+      getAttributesFrom(start).flatMap(attrs =>
+        if (attrs.size < 50) Future.successful(as ++ attrs) else go(start + 50)(as ++ attrs)
+      )
+
+    go(0)(Nil)
+  }
```

Comment:
The implementation should be (and is already) in the `Impl` class
```suggestion
```

Reply:
I didn't understand